lib/paper_trail/version_concern.rb

```diff
@@ -417,7 +417,14 @@ def reify_has_many_through(associations, model, options = {})
       associations.each do |assoc|
         next unless assoc.klass.paper_trail_enabled_for_model?
         through_collection = model.send(assoc.options[:through])
-        collection_keys = through_collection.map { |through_model| through_model.send(assoc.foreign_key) }
+
+        # if the association is a has_many association again, then call reify_has_manys for each through_collection
+        if !assoc.source_reflection.belongs_to? && through_collection.present?
+          through_collection.each { |through_model| reify_has_manys(through_model,options) }
+          return
```

Comment:
I guess my question / thought is, shouldn't the method still perform the same behavior it does prior to the return in these cases? In the normal / existing method [flow we assign the new collection of updated issues to the association definition](https://github.com/theRealNG/paper_trail/blob/issue%23590/lib/paper_trail/version_concern.rb#L458) and return it in one call... In this case is it not necessary to perform that since the collection is already assigned to the association, or am I missing something here?